mvvm_cards/src/shared/Modal.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from "react";
```

Comment:
To be honest the naming of such Components alway was some funny things for me. Some people say that everything that appears like a Popup is a Model, some people say that if the Popup has some Actions that user must execute (eg. Save & Exit) and if the background is blocking - so it's a Dialog.

I prefer to follow next:
1) a Modal is a Component that appears outside of the document flow (like an independent component). It could be some Additional Info, some text, image, etc.
2) a Dialog is a Subset of the Modal. It should also appear outside of the document but in addition of the Modal it should have some actions that user should execute to do something with the dialog and the document background should be disabled

https://material-ui.com/components/dialogs/

![image](https://user-images.githubusercontent.com/9897782/72661321-1e7aa380-39e1-11ea-9bdf-3fc2d98704f2.png)

Reply:
Renamed it to ModalDialog

Reply:
FYI - In WPF and other windows frameworks, a dialog is a little child window that floats above the other UI.   Dialogs can be either modal or modeless.  Modal means that the UI behind the dialog is paused while the dialog is open.  Modeless means that the dialog UI operates in parallel with the main application UI.   So a save dialog would typically be modal and something like a tool selector dialog would typically be modeless.

Specifically, there is no concept of a window class that is specifically one or the other.  The programmer chooses to open any window as modal or modeless, depending on the need.   Not sure if createPortal supports this concept.   Also not sure the best way to create something modeless. Maybe window.open?